android-core/src/main/java/com/mparticle/internal/MParticleApiClientImpl.java

```diff
@@ -203,28 +207,29 @@ public void fetchConfig(boolean force) throws IOException, MPConfigException {
         }
     }
 
-    public JSONObject fetchAudiences() {
-
-        JSONObject response = null;
+    public void fetchUserAudience(BaseAudienceTask task, long mpId) {
+        JSONObject jsonResponse = null;
         try {
-            Logger.debug("Starting Segment Network request");
-            MPConnection connection = getUrl(Endpoint.AUDIENCE).openConnection();
-            connection.setConnectTimeout(mConfigManager.getConnectionTimeout());
-            connection.setReadTimeout(mConfigManager.getConnectionTimeout());
+            MPConnection connection = getUrl(Endpoint.AUDIENCE,mpId).openConnection();
+            Logger.verbose("Audience API request: \n" + connection.getURL().toString());
             connection.setRequestProperty("User-Agent", mUserAgent);
-
             addMessageSignature(connection, null);
+            connection.setRequestProperty("x-mp-key", mApiKey);
             makeUrlRequest(Endpoint.AUDIENCE, connection, true);
             if (connection.getResponseCode() == HttpURLConnection.HTTP_FORBIDDEN) {
-                Logger.error("Segment call forbidden: is Segmentation enabled for your account?");
+                Logger.error("Audience API call forbidden: please verify your workspace is enabled for audiences");
```

Comment:
I'd like to align all of our error messages.  Since iOS and Android aren't out yet, but Web is, let's align to Web's error message:
```suggestion
                Logger.error("Your workspace is not enabled to retrieve user audiences");
```